pkg/server/server.go

```diff
@@ -26,15 +27,17 @@ import (
 	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/trace"
+	"golang.org/x/exp/slices"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/metadata"
 )
 
 type ExperimentalFeatureFlag string
 
 const (
-	AuthorizationModelIDHeader = "openfga-authorization-model-id"
-	authorizationModelIDKey    = "authorization_model_id"
+	AuthorizationModelIDHeader  = "openfga-authorization-model-id"
+	authorizationModelIDKey     = "authorization_model_id"
+	ExperimentalCheckQueryCache = "check-query-cache"
```

Comment:
Why the `Experimental` prefix?

Reply:
This should be a typed enum const so that it can be treated as an enum.

```go
CheckQueryCache ExperimentalFeatureFlag = "check-query-cache"
```